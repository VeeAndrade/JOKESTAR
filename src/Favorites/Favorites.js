import React from 'react';
import './Favorites.scss';
import FavoriteJoke from '../FavoriteJoke/FavoriteJoke'
import { useSelector } from 'react-redux';

const Favorites = () => {
  const favorites = useSelector(state => state.favorites)
  let favoritesArray = favorites.map(favorite => <FavoriteJoke jokeObj={favorite}/>)
  return (
    <section className='favorites-container'>
      {favoritesArray}
    </section>
  )
}

export default Favorites;