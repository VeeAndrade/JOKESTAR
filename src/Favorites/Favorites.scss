.favorites-container {
  border: solid 2px #fff;
  overflow: scroll;
  height: 80%;
  width: 95%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}