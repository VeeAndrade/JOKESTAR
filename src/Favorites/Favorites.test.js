import React from 'react';
import { Favorites, mapStateToProps } from './Favorites';
import { shallow } from 'enzyme';

describe('FavoriteJoke', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow( <Favorites 
    favorites={[{id: 42, setup: 'how much does a polarbear weigh?', delivery: 'enough to bresk the ice'}]}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
}); 

describe('mapStateToProps', () => {
  it('should return an array of favorites', () => {
    const mockState = {
      username: 'EvilKitty',
      favorites: [{id: 42, setup: 'setup joke', delivery: 'deliver'}]
    }
    const expected = {
      favorites: [{
        id: 42,
        setup: 'setup joke',
        delivery: 'deliver'
      }]
    }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
  })
})