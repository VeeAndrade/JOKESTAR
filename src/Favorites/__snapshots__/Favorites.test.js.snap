// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteJoke should match the snapshot 1`] = `
<section
  className="favorites-container"
>
  <FavoriteJoke
    jokeObj={
      Object {
        "delivery": "enough to bresk the ice",
        "id": 42,
        "setup": "how much does a polarbear weigh?",
      }
    }
    key="42"
  />
</section>
`;
